File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/AffiliateTrafficRow.cpp


#include "AffiliateTrafficRow.h"

using namespace Tiny;

AffiliateTrafficRow::AffiliateTrafficRow()
{
	traffic_id = std::string();
	traffic_ip = std::string();
	traffic_url = std::string();
	traffic_affiliate = std::string();
	traffic_referrer = std::string();
	traffic_timestamp = std::string();
}

AffiliateTrafficRow::AffiliateTrafficRow(std::string jsonString)
{
	this->fromJson(jsonString);
}

AffiliateTrafficRow::~AffiliateTrafficRow()
{

}

void
AffiliateTrafficRow::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *traffic_idKey = "traffic_id";

    if(object.has_key(traffic_idKey))
    {
        bourne::json value = object[traffic_idKey];



        jsonToValue(&traffic_id, value, "std::string");


    }

    const char *traffic_ipKey = "traffic_ip";

    if(object.has_key(traffic_ipKey))
    {
        bourne::json value = object[traffic_ipKey];



        jsonToValue(&traffic_ip, value, "std::string");


    }

    const char *traffic_urlKey = "traffic_url";

    if(object.has_key(traffic_urlKey))
    {
        bourne::json value = object[traffic_urlKey];



        jsonToValue(&traffic_url, value, "std::string");


    }

    const char *traffic_affiliateKey = "traffic_affiliate";

    if(object.has_key(traffic_affiliateKey))
    {
        bourne::json value = object[traffic_affiliateKey];



        jsonToValue(&traffic_affiliate, value, "std::string");


    }

    const char *traffic_referrerKey = "traffic_referrer";

    if(object.has_key(traffic_referrerKey))
    {
        bourne::json value = object[traffic_referrerKey];



        jsonToValue(&traffic_referrer, value, "std::string");


    }

    const char *traffic_timestampKey = "traffic_timestamp";

    if(object.has_key(traffic_timestampKey))
    {
        bourne::json value = object[traffic_timestampKey];



        jsonToValue(&traffic_timestamp, value, "std::string");


    }


}

bourne::json
AffiliateTrafficRow::toJson()
{
    bourne::json object = bourne::json::object();





    object["traffic_id"] = getTrafficId();






    object["traffic_ip"] = getTrafficIp();






    object["traffic_url"] = getTrafficUrl();






    object["traffic_affiliate"] = getTrafficAffiliate();






    object["traffic_referrer"] = getTrafficReferrer();






    object["traffic_timestamp"] = getTrafficTimestamp();



    return object;

}

std::string
AffiliateTrafficRow::getTrafficId()
{
	return traffic_id;
}

void
AffiliateTrafficRow::setTrafficId(std::string traffic_id)
{
	this->traffic_id = traffic_id;
}

std::string
AffiliateTrafficRow::getTrafficIp()
{
	return traffic_ip;
}

void
AffiliateTrafficRow::setTrafficIp(std::string traffic_ip)
{
	this->traffic_ip = traffic_ip;
}

std::string
AffiliateTrafficRow::getTrafficUrl()
{
	return traffic_url;
}

void
AffiliateTrafficRow::setTrafficUrl(std::string traffic_url)
{
	this->traffic_url = traffic_url;
}

std::string
AffiliateTrafficRow::getTrafficAffiliate()
{
	return traffic_affiliate;
}

void
AffiliateTrafficRow::setTrafficAffiliate(std::string traffic_affiliate)
{
	this->traffic_affiliate = traffic_affiliate;
}

std::string
AffiliateTrafficRow::getTrafficReferrer()
{
	return traffic_referrer;
}

void
AffiliateTrafficRow::setTrafficReferrer(std::string traffic_referrer)
{
	this->traffic_referrer = traffic_referrer;
}

std::string
AffiliateTrafficRow::getTrafficTimestamp()
{
	return traffic_timestamp;
}

void
AffiliateTrafficRow::setTrafficTimestamp(std::string traffic_timestamp)
{
	this->traffic_timestamp = traffic_timestamp;
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/AffiliateTrafficRow.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Affiliate Web Traffic Entry
 */
@Schema(description = "Affiliate Web Traffic Entry")
@Validated
@NotUndefined



public class AffiliateTrafficRow   {
  @JsonProperty("traffic_id")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficId = null;

  @JsonProperty("traffic_ip")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficIp = null;

  @JsonProperty("traffic_url")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficUrl = null;

  @JsonProperty("traffic_affiliate")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficAffiliate = null;

  @JsonProperty("traffic_referrer")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficReferrer = null;

  @JsonProperty("traffic_timestamp")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String trafficTimestamp = null;


  public AffiliateTrafficRow trafficId(String trafficId) { 

    this.trafficId = trafficId;
    return this;
  }

  /**
   * Get trafficId
   * @return trafficId
   **/
  
  @Schema(description = "")
  
  public String getTrafficId() {  
    return trafficId;
  }



  public void setTrafficId(String trafficId) { 
    this.trafficId = trafficId;
  }

  public AffiliateTrafficRow trafficIp(String trafficIp) { 

    this.trafficIp = trafficIp;
    return this;
  }

  /**
   * Get trafficIp
   * @return trafficIp
   **/
  
  @Schema(description = "")
  
  public String getTrafficIp() {  
    return trafficIp;
  }



  public void setTrafficIp(String trafficIp) { 
    this.trafficIp = trafficIp;
  }

  public AffiliateTrafficRow trafficUrl(String trafficUrl) { 

    this.trafficUrl = trafficUrl;
    return this;
  }

  /**
   * Get trafficUrl
   * @return trafficUrl
   **/
  
  @Schema(description = "")
  
  public String getTrafficUrl() {  
    return trafficUrl;
  }



  public void setTrafficUrl(String trafficUrl) { 
    this.trafficUrl = trafficUrl;
  }

  public AffiliateTrafficRow trafficAffiliate(String trafficAffiliate) { 

    this.trafficAffiliate = trafficAffiliate;
    return this;
  }

  /**
   * Get trafficAffiliate
   * @return trafficAffiliate
   **/
  
  @Schema(description = "")
  
  public String getTrafficAffiliate() {  
    return trafficAffiliate;
  }



  public void setTrafficAffiliate(String trafficAffiliate) { 
    this.trafficAffiliate = trafficAffiliate;
  }

  public AffiliateTrafficRow trafficReferrer(String trafficReferrer) { 

    this.trafficReferrer = trafficReferrer;
    return this;
  }

  /**
   * Get trafficReferrer
   * @return trafficReferrer
   **/
  
  @Schema(description = "")
  
  public String getTrafficReferrer() {  
    return trafficReferrer;
  }



  public void setTrafficReferrer(String trafficReferrer) { 
    this.trafficReferrer = trafficReferrer;
  }

  public AffiliateTrafficRow trafficTimestamp(String trafficTimestamp) { 

    this.trafficTimestamp = trafficTimestamp;
    return this;
  }

  /**
   * Get trafficTimestamp
   * @return trafficTimestamp
   **/
  
  @Schema(description = "")
  
  public String getTrafficTimestamp() {  
    return trafficTimestamp;
  }



  public void setTrafficTimestamp(String trafficTimestamp) { 
    this.trafficTimestamp = trafficTimestamp;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AffiliateTrafficRow affiliateTrafficRow = (AffiliateTrafficRow) o;
    return Objects.equals(this.trafficId, affiliateTrafficRow.trafficId) &&
        Objects.equals(this.trafficIp, affiliateTrafficRow.trafficIp) &&
        Objects.equals(this.trafficUrl, affiliateTrafficRow.trafficUrl) &&
        Objects.equals(this.trafficAffiliate, affiliateTrafficRow.trafficAffiliate) &&
        Objects.equals(this.trafficReferrer, affiliateTrafficRow.trafficReferrer) &&
        Objects.equals(this.trafficTimestamp, affiliateTrafficRow.trafficTimestamp);
  }

  @Override
  public int hashCode() {
    return Objects.hash(trafficId, trafficIp, trafficUrl, trafficAffiliate, trafficReferrer, trafficTimestamp);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AffiliateTrafficRow {\n");
    
    sb.append("    trafficId: ").append(toIndentedString(trafficId)).append("\n");
    sb.append("    trafficIp: ").append(toIndentedString(trafficIp)).append("\n");
    sb.append("    trafficUrl: ").append(toIndentedString(trafficUrl)).append("\n");
    sb.append("    trafficAffiliate: ").append(toIndentedString(trafficAffiliate)).append("\n");
    sb.append("    trafficReferrer: ").append(toIndentedString(trafficReferrer)).append("\n");
    sb.append("    trafficTimestamp: ").append(toIndentedString(trafficTimestamp)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
